test/compiler/inference.jl

```diff
@@ -4250,3 +4250,9 @@ let # NOTE make sure this toplevel chunk doesn't contain any local binding
     while xcond end
 end
 @test !xcond
+
+struct Issue45780
+    oc::Core.OpaqueClosure{Tuple{}}
+end
+f45780() = Val{Issue45780(@Base.Experimental.opaque ()->1).oc()}()
+@test (@inferred @code_typed f45780()) == Val{1}()
```

Comment:
```suggestion
@test only(Base.return_types(f45780)) == Val{1}()
```

Reply:
That `@inferred` essentially checks if the return type-stability of `code_typed`, but not that of `f45780`.
Also note that `@code_typed` returns an IR too.

Reply:
oops, I meant to check `@test (@inferred f45780()) == Val{1}()`. Would that be the same?

Reply:
Yeah, your option is better as it also includes actual execution also.